omod/src/test/java/org/openmrs/module/bedmanagement/rest/resource/AdmissionLocationResourceTest.java

```diff
@@ -89,4 +92,138 @@ public void shouldGetWardAlongWithTotalBedsAndOccupiedBedsWhenTheRepresentionIsD
         assertEquals(6, orthopaedicWard.get("totalBeds"));
         assertEquals(2, orthopaedicWard.get("occupiedBeds"));
     }
+
+    @Test
+    public void shouldReturnAllAdmissionLocations() throws Exception {
+        MockHttpServletRequest request = request(RequestMethod.GET, getURI());
+        SimpleObject object = deserialize(handle(request));
+        List results = (ArrayList) object.get("results");
+        LinkedHashMap location1 = (LinkedHashMap) results.get(0);
+        LinkedHashMap ward1 = (LinkedHashMap) location1.get("ward");
+        LinkedHashMap location2 = (LinkedHashMap) results.get(1);
+        LinkedHashMap ward2 = (LinkedHashMap) location2.get("ward");
+        LinkedHashMap location3 = (LinkedHashMap) results.get(2);
+        LinkedHashMap ward3 = (LinkedHashMap) location3.get("ward");
+
+        Assert.assertEquals(3, results.size());
+        Assert.assertEquals("7779d653-393b-4118-9c83-a3715b82d4ac", ward1.get("uuid"));
+        Assert.assertTrue(location1.containsKey("ward"));
+        Assert.assertTrue(location1.containsKey("totalBeds"));
+        Assert.assertTrue(location1.containsKey("occupiedBeds"));
+        Assert.assertEquals("e26cea2c-1b9f-666e-6511-f3ef6c88af6f", ward2.get("uuid"));
+        Assert.assertEquals("19e023e8-20ee-4237-ade6-9e68f897b7a9", ward3.get("uuid"));
+    }
+
+    @Test
+    public void shouldReturnAdmissionLocationByUuidWithFullRepresentation() throws Exception {
+        MockHttpServletRequest request = request(RequestMethod.GET, getURI() + "/" + getUuid());
+        request.setParameter("v", "full");
+        SimpleObject location = deserialize(handle(request));
+        LinkedHashMap ward = location.get("ward");
+
+        Assert.assertEquals("19e023e8-20ee-4237-ade6-9e68f897b7a9", ward.get("uuid"));
+        Assert.assertEquals(6, location.get("totalBeds"));
+        Assert.assertEquals(2, location.get("occupiedBeds"));
+        Assert.assertTrue(location.containsKey("bedLayouts"));
+        List bedLayouts = (ArrayList) location.get("bedLayouts");
+        Assert.assertEquals(6, bedLayouts.size());
+
+        MockHttpServletRequest request2 = request(RequestMethod.GET, getURI() + "/98bc9b32-9d1a-11e2-8137-0800271c1b75");
+        SimpleObject location2 = deserialize(handle(request2));
+        LinkedHashMap wardRoom = location2.get("ward");
```

Comment:
@sanishmaharjan rather than navigating `LinkedHashMap`s you could use the convenient Apache bean utility class `PropertyUtils` such as [here](https://github.com/openmrs/openmrs-module-webservices.rest/blob/023de3cd05374e240ac4b21de19e04defa0b357f/omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/LocationController1_9Test.java#L122). This will help you nicely dig into any JSON structured response.